validator/valnode/validation_api.go

```diff
@@ -80,12 +80,12 @@ func NewExecutionServerAPI(valSpawner validator.ValidationSpawner, execution val
 	}
 }
 
-func (a *ExecServerAPI) CreateExecutionRun(ctx context.Context, wasmModuleRoot common.Hash, jsonInput *server_api.InputJSON) (uint64, error) {
+func (a *ExecServerAPI) CreateExecutionRun(ctx context.Context, wasmModuleRoot common.Hash, jsonInput *server_api.InputJSON, useBoldMachine *bool) (uint64, error) {
```

Comment:
Perhaps the best solution to avoid putting pointers everywhere would be to have this line be the only place we use a `*bool` for useBoldMachine, and have it default to false if nil or otherwise dereference it before calling `a.execSpawner.CreateExecutionRun`